katas/pete-the-baker/pete-the-baker.js

```diff
@@ -1,48 +1,30 @@
 // Pete likes to bake some cakes. He has some recipes and ingredients. Unfortunately he is not good in maths. Can you help him to find out, how many cakes he could bake considering his recipes?
 // Write a function cakes(), which takes the recipe (object) and the available ingredients (also an object) and returns the maximum number of cakes Pete can bake (integer). For simplicity there are no units for the amounts (e.g. 1 lb of flour or 200 g of sugar are simply 1 or 200). Ingredients that are not present in the objects, can be considered as 0.
 // Examples:
-// // must return 0
-cakes({flour: 500, sugar: 200, eggs: 1}, {flour: 1000, sugar: 1200, eggs: 5, milk: 200}); 
-
 
 // // must return 0
-// cakes({apples: 3, flour: 300, sugar: 150, milk: 100, oil: 100}, {sugar: 500, flour: 2000, milk: 2000});
+cakes({apples: 3, flour: 300, sugar: 150, milk: 100, oil: 100}, {sugar: 500, flour: 2000, milk: 2000});
 // must return 0
-// cakes({apples: 3, flour: 300, sugar: 150, milk: 100, oil: 100}, {sugar: 500, flour: 2000, milk: 2000});
-
-// cakes({flour: 500}, {flour: 1000}); 
-
-function cakes(recipe, available){
+cakes({apples: 3, flour: 300, sugar: 150, milk: 100, oil: 100}, {sugar: 500, flour: 2000, milk: 2000});
+// must return 2
+cakes({flour: 500, sugar: 200, eggs: 1}, {flour: 1000, sugar: 1200, eggs: 5, milk: 200}); 
 
-  //declare empty arrays for comparison and interger manipulation
-  let recipeArr = []
-  let availableArr = []
+function cakes (recipe, avaiable){
+  // store the keys as a new Array in order to compare them
+  let recipeArr = Object.keys(recipe)
+  let availableArr = Object.keys(avaiable)
+  // storage of quotients when dividing avaiable/recipe
   let howManyCakes = []
-
-    //push keys of recipe and avaiable into empty arrays
-    for (let key in recipe) {
-      // console.log(`${key} in ${obj1[key]}`)
-      recipeArr.push(key)
-    }
-    for (let key in available) {
-      availableArr.push(key)
-    }
-
     //check to see if avaiable includes recipe
     if(!availableArr.includes(...recipeArr)){
```

Comment:
Hi Colby! Nice job cleaning this up. I have a question for you about this line. What would happen if you took the recipe with apples and put apples in the second position, instead of first?

Reply:
do you mean like this? 
cakes({flour: 300, apples: 3,  sugar: 150, milk: 100, oil: 100}, { apples: 3, flour: 300,  sugar: 150, milk: 100, oil: 100});
I ran this example and it returns 1 when apple is in the first position and the second position as well. In this example, since what is available is also included in recipe we can run the function and the apples will still divide with one another no matter what position they're in.

Reply:
You're right, it would work for that case. What about the first example you have, at the top of this file? The recipe has apples, but they aren't in the available ingredients.

What does should this do?
```js
cakes(
  {flour: 300, apples: 3,  sugar: 150, milk: 100, oil: 100},
  {sugar: 500, flour: 2000, milk: 2000}
)
```

I think there's some interesting stuff happening with `.includes` here.

Reply:
ohhhhhhh ohhhh my. This returns 3, but it shouldn't! It should return 0. What's in the recipe isn't available. oh boy.. hahahah this .includes() may not be the way to go. in order to get it to work, i had to add another condition.. but this leaves me thinking that there must be another method that can take care of this.. for now, i put it this way 

``` if(availableArr.includes(...recipeArr) && recipeArr.length === availableArr.length){
      
         for (let val in available) {
       
           howManyCakes.push(Math.floor(available[val]/recipe[val]))
        }
      } else { return 0 }```